pandas/tests/indexing/test_loc.py

```diff
@@ -2666,6 +2674,12 @@ def test_loc_getitem_series_label_list_missing_integer_values(self):
         with pytest.raises(KeyError, match="not in index"):
             ser.loc[np.array([9730701000001104, 10047311000001102])]
 
+    def test_loc_set_item_empty_list(self):
+        # GH: 29707
+        df = DataFrame({"a": [2, 3]})
+        df.loc[[]] = 0.1
+        assert df.a.dtype == "int64"
```

Comment:
Can you compare the whole DataFrame?

like:

```
df = DataFrame({"a": [2, 3]})
expected = df.copy()
df.loc[[]] = 0.1
tm.assert_frame_equal(df, expected)
```

Same above. otherwise lgtm

Reply:
That helper function is certainly useful. Thanks for the suggestion!